src/jit/importer.cpp

```diff
@@ -8707,24 +8707,13 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
         // imperative security in the method. This is to give security a
         // chance to do any setup in the prolog and cleanup in the epilog if needed.
 
-        if (compIsForInlining())
-        {
-            // Cannot handle this if the method being imported is an inlinee by itself.
-            // Because inlinee method does not have its own frame.
-
-            compInlineResult->NoteFatal(InlineObservation::CALLEE_NEEDS_SECURITY_CHECK);
-            return TYP_UNDEF;
-        }
-        else
-        {
-            tiSecurityCalloutNeeded = true;
+        tiSecurityCalloutNeeded = true;
 
-            // If the current method calls a method which needs a security check,
-            // (i.e. the method being compiled has imperative security)
-            // we need to reserve a slot for the security object in
-            // the current method's stack frame
-            opts.compNeedSecurityCheck = true;
-        }
+        // If the current method calls a method which needs a security check,
+        // (i.e. the method being compiled has imperative security)
+        // we need to reserve a slot for the security object in
+        // the current method's stack frame
+        opts.compNeedSecurityCheck = true;
```

Comment:
Out of curiosity, is `compNeedSecurityCheck` still actually used? Can the whole code actually be hit?

Looks like both CoreRT and CoreCLR set `CORINFO_FLG_NOSECURITYWRAP` on every method and never set `CORINFO_FLG_SECURITYCHECK`. I found two places where `compNeedSecurityCheck` is set to `true` and both are guarded by checks on these flags.

Reply:
The JIT code is shared with the .NET Framework so security/verification code is likely still needed. Though perhaps it could be ifdefed out...

Reply:
Ah, I was quite convinced that I saw somewhere that the sharing is no longer happening. If that's the case then maybe it would make sense to wrap it under some `FEATURE_` flag.

Reply:
We still mirror all the jit changes back to a full framework build. We get extra test coverage on full framework that we just can't replicate on core.

I would not be in favor of introducing new ifdefs here unless they clearly enhance readability or offer some significant jit throughput benefit.

Reply:
Understood. I just wasn't sure whether the code was actually dead or not. Thanks for clearing it up.